ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -79,7 +88,16 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
       val canUserSubmitData = userRepository.canUserSubmitData()
 
       // Handle collect button clicks
-      adapter.setCollectDataListener { onCollectData(canUserSubmitData, hasValidTasks(it), it) }
+      adapter.setCollectDataListener {
+        val job =
+          lifecycleScope.launch {
+            mapContainerViewModel.activeSurveyDataSharingTermsFlow.cancellable().collectLatest {
+              hasDataSharingTerms ->
+              onCollectData(canUserSubmitData, hasValidTasks(it), hasDataSharingTerms, it)
```

Comment:
Can we name the variable explicitly instead of `it` since it's getting harder to read due to too many lines?